config/configcompression/compressiontype.go

```diff
@@ -25,19 +37,45 @@ func (ct *Type) IsCompressed() bool {
 	return *ct != typeEmpty && *ct != typeNone
 }
 
-func (ct *Type) UnmarshalText(in []byte) error {
-	typ := Type(in)
-	if typ == TypeGzip ||
-		typ == TypeZlib ||
-		typ == TypeDeflate ||
-		typ == TypeSnappy ||
-		typ == TypeZstd ||
-		typ == TypeLz4 ||
-		typ == typeNone ||
-		typ == typeEmpty {
-		*ct = typ
+func (ct *TypeWithLevel) UnmarshalText(in []byte) error {
+	var err error
+	parts := strings.Split(string(in), "/")
+	compressionTyp := Type(parts[0])
+	level := zlib.DefaultCompression
+	if len(parts) == 2 {
+		level, err = strconv.Atoi(parts[1])
+		if err != nil {
+			return fmt.Errorf("invalid compression level: %q", parts[1])
+		}
+		if compressionTyp == TypeSnappy ||
+			compressionTyp == TypeLz4 ||
+			compressionTyp == typeNone ||
+			compressionTyp == typeEmpty {
+			return fmt.Errorf("compression level is not supported for %q", compressionTyp)
+		}
+	}
+	ct.Level = Level(level)
+	if (compressionTyp == TypeGzip && isValidLevel(level)) ||
+		(compressionTyp == TypeZlib && isValidLevel(level)) ||
+		(compressionTyp == TypeDeflate && isValidLevel(level)) ||
+		compressionTyp == TypeSnappy ||
+		compressionTyp == TypeLz4 ||
+		compressionTyp == TypeZstd ||
+		compressionTyp == typeNone ||
+		compressionTyp == typeEmpty {
+		ct.Level = Level(level)
+		ct.Type = compressionTyp
 		return nil
 	}
-	return fmt.Errorf("unsupported compression type %q", typ)
 
+	return fmt.Errorf("unsupported compression type/level %s/%d", compressionTyp, ct.Level)
```

Comment:
This would be confusing output if user didn't provide level

Reply:
this should print the default level, however I can remove it since this line is only reached when the compression level isn't specified.

Reply:
I made a change to the error message, please let me know if you think this is clearer

https://github.com/open-telemetry/opentelemetry-collector/pull/11084/commits/c7f36ad44861a58491a09f537207a20decb3c186